src/scaling.jl

```diff
@@ -0,0 +1,192 @@
+"""
+    Scaling <: Transform
+
+Linearly scale the data as `ax + b`, according to some statistics `a` and `b`.
+"""
+abstract type Scaling <: Transform end
+
+"""
+    IdentityScaling
+
+Represents the no-op scaling which simply returns the data it is applied on.
+"""
+struct IdentityScaling <: Scaling end
+# Convenience method
+IdentityScaling(args...; kwargs...) = IdentityScaling()
+
+apply!(A::AbstractArray{T}, scaling::IdentityScaling; kwargs...) where T <: Real = A
+
+"""
+    MeanStdScaling(mean, std) <: Scaling
+
+Linearly scale the data by a statistical `mean` and standard deviation `std`.
+The first `mean` and `std` calculated on some data will remain fixed.
```

Comment:
Important to take note of this throughout.

Reply:
I think this could be better phrased?

```
Once computed, the statistics of a `MeanStdScaling` are immutable.
```

Reply:
Can we compute them at construction time instead of at runtime? 

`t = MeanStdScaling(data)` that would calculate the mean and std that would me used anything aply(x, t) the previously computed ones would be used? 

That would enable us to re-use the general apply methods.

Reply:
> Can we compute them at construction time instead of at runtime?

Good point. I made an additional constructor to do that, but I think it should be the default constructor.

Reply:
> That would enable us to re-use the general apply methods.

The only problem I see with that is passing the `name` keyword argument to `_apply!`, which ensures the slices/columns are scaled in the correct order.

Reply:
Passing a name kwarg seems fine to me